skyportal/handlers/api/gcn_tach.py

```diff
@@ -0,0 +1,333 @@
+              pageInfo{{
+                hasNextPage
+                hasPreviousPage
+                startCursor
+                endCursor
+              }}
+              edges {{
+                node {{
+                  id
+                  id_
+                  received
+                  subject
+                  evtidCircular{{
+                    event
+                  }}
+                  cid
+                  evtid
+                  oidCircular{{
+                    telescope
+                    detector
+                    oidEvent{{
+                      wavelength
+                      messenger
+                    }}
+                  }}
+                  }}
+                }}
+              }}
+          }}'''
+    }
+    headers = {
+        "Content-Type": "application/json",
+        "Origin": "https://heasarc.gsfc.nasa.gov",
+    }
+
+    response = requests.request("POST", url, json=payload, headers=headers)
+
+    circulars = gcn_event.circulars
+    # this is a dict with key = circular id and value = title
+
+    aliases = []
+    if response.status_code == 200:
+        data = response.json()
+        if data["data"]["allCirculars"]["totalCount"] > 0:
+            events = data["data"]["allCirculars"]["edges"]
+            aliases.append(events[0]["node"]["evtidCircular"]["event"].replace(" ", ""))
+            new_circulars = {}
+            for event in events:
+                #     aliases.append(event["node"]["evtidCircular"]["event"].replace(" ", ""))
+                if event["node"]["id_"] not in circulars.keys():
+                    new_circulars[event["node"]["id_"]] = event["node"]["subject"]
+            day = re.sub(r'\D', '', aliases[0])
+            aliases = get_aliases(list(new_circulars.keys()), day)
+            return aliases, new_circulars
+        else:
+            return [], {}
+    return [], {}
+
+
+def post_aliases(dateobs, tach_id, user, push_all):
+    with Session() as session:
+        stmt = GcnEvent.select(user).where(GcnEvent.dateobs == dateobs)
+        gcn_event = session.scalars(stmt).first()
+        if gcn_event is None:
+            return
+        gcn_event.tach_id = tach_id
+        new_gcn_aliases, new_gcn_circulars = get_tach_event_aliases(tach_id, gcn_event)
+
+        if len(new_gcn_aliases) == 0:
+            return
+
+        if gcn_event.aliases is None:
+            gcn_event.aliases = new_gcn_aliases
+        else:
+            gcn_aliases = [alias.upper() for alias in gcn_event.aliases]
+            for new_gcn_alias in new_gcn_aliases:
+                if new_gcn_alias.upper() not in gcn_aliases:
+                    gcn_aliases.append(new_gcn_alias)
+            gcn_event.aliases = gcn_aliases
+
+        if gcn_event.circulars is None:
+            gcn_event.circulars = new_gcn_circulars
+        else:
+            gcn_event.circulars = {**gcn_event.circulars, **new_gcn_circulars}
+
+        session.commit()
+
+    push_all(
```

Comment:
This can just be imported from flow. I worry about carrying this dependency on the handler itself.